### ISSUE TITLE
Malformed SimpleITK version when built from the source tarball

### ISSUE THREAD

**Issue by ghisvail** [SimpleITK/SimpleITK#203]
```
CMake Warning at CMake/sitkSourceVersion.cmake:56 (message):
  git tag: "GIT-NOTFOUND" does not match expected version format!
Call Stack (most recent call first):
  Version.cmake:17 (include)
  CMakeLists.txt:25 (include)


-- Building SimpleITK version "1.0.0-gGIT-N"
```

I'd expect the version to just be `1.0.0` when `git` is not found.

**Comment by blowekamp** [comment]
Are you using the tar ball from source forge or the manually packaged one uploaded to GitHub:?
https://sourceforge.net/projects/simpleitk/files/SimpleITK/1.0.0/Source/
https://github.com/SimpleITK/SimpleITK/releases/download/v1.0.0/SimpleITK-1.0.0.tar.gz

Are you setting `SimpleITK_BUILD_DISTRIBUTE:BOOL=ON`? This will remove the `-ghash` from the version. Although I'm not sure what happens with the "GIT-NOTFOUND` situation.

There is no way to disable the "automatic" GitHub archive, which produces this problem. This is becoming a more common problem. As the GitHub auto archive can be produced from any commit, a warning will be produces as the configuration can not determine where in the repositories history the current source is. I'll add code to handle it better this case, but it is still going to produce a warning.

**Comment by ghisvail** [comment]
> Are you using the tar ball from source forge or the manually packaged one uploaded to GitHub:?

I guess if you're asking the question, then they must be different. Right now, I am using the SourceForge ones.

> Are you setting SimpleITK_BUILD_DISTRIBUTE:BOOL=ON?

You are correct, I am not.

**Comment by blowekamp** [comment]
The difference between the proper SourceForge version ( manual package ), and the automatic GitHub is that we add a file `CMake/sitkSourceVersionVars.cmake` I verified the three source balls on SourceForge all contain that file with the following contents:
```
macro( _set_if_not_empty var value )
  if( NOT "${value}" STREQUAL "" )
    set( ${var} "${value}" )
  endif()
endmacro()

set( _GIT_VERSION_MAJOR "1" )
set( _GIT_VERSION_MINOR "0" )
_set_if_not_empty( _GIT_VERSION_PATCH "0" )
_set_if_not_empty( _GIT_VERSION_TWEAK "" )
_set_if_not_empty( _GIT_VERSION_RC "" )
_set_if_not_empty( _GIT_VERSION_POST "" )
_set_if_not_empty( _GIT_VERSION_DEV "" )
_set_if_not_empty( _GIT_VERSION_HASH "2eac9" )
```

This file is automatically generated when building from the git repository, extracting a descriptive version for development based on the number of commits from a tag, and including the hash of the current commit. When an source archive is created the properly configured file is included.

I tested a compiling a Superbuild of SimpleITK from the download tar.gz, this file was properly detected and the following line was produced during configuration:
```
-- Building SimpleITK version "1.0.0-g2eac9"
```

So I am not able to reproduce this issue.
